pyproject.toml

```diff
@@ -1,56 +1,93 @@
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: Implementation :: CPython",
+    "Topic :: Scientific/Engineering",
+    "Topic :: Scientific/Engineering :: Physics",
+]
+dependencies = [
+    "qastle>=0.17",
+    "func_adl>=3.2.6",
+    "requests>=2.31",
+    "pydantic>=1.10,<2.0",  # FIXME: Adopt pydantic 2.0 API
+    "httpx>=0.24",
+    "miniopy-async>=1.15",
+    "tinydb>=4.7",
+    "google-auth>=2.17",
+    "typer[all]>=0.9.0",
+    "PyYAML>=6.0",
+    "types-PyYAML>=6.0",
+    "rich>=13.0.0",  # databinder
+    "nest-asyncio>=1.5.7",  # databinder # FIXME: nest-asyncio deprecated
+    "aiofile",  # compatible versions controlled through miniopy-async
+    "make-it-sync",  # compatible versions controlled through func_adl
 ]
-readme = "README.md"
-packages = [{ include = "servicex" }]
-
-[tool.poetry.scripts]
-servicex = "servicex.app.main:app"
-
-[tool.poetry.dependencies]
-python = "^3.8, < 3.12"
-qastle = "^0.16"
-func_adl = "^3.2.6"
-requests = "^2.31"
-pydantic = "^1.10"
-httpx = "^0.24"
-miniopy-async = "^1.15"
-tinydb = "^4.7"
-google-auth = "^2.17"
-typer = { extras = ["all"], version = "^0.9.0" }
-PyYAML = "^6.0"
-types-PyYAML = "^6.0"
-make_it_sync = "^1.0.0"
-
-pandas = { version = "^2.0.2", optional = true }
-pyarrow = { version = "^12.0.0", optional = true }
-fastparquet = { version = "^2023.4.0", optional = true }
-
-sphinx = { version = "^7.0.1", optional = true }
-furo = { version = "^2023.5.20", optional = true }
-jupyter = "^1.0.0"
-
-
-[tool.poetry.group.test.dependencies]
-pytest = "^7.2.0"
-pytest-mock = "^3.10.0"
-pytest-cov = "^4.0.0"
-flake8 = "^5.0.4"
-mypy = "^0.981"
-pytest-asyncio = "^0.21.0"
-asyncmock = "^0.4.2"
 
+[project.urls]
+Documentation = "https://github.com/ssl-hep/ServiceX_frontend"
+Homepage = "https://github.com/ssl-hep/ServiceX_frontend"
+"Issue Tracker" = "https://github.com/ssl-hep/ServiceX_frontend/issues"
+"Release Notes" = "https://github.com/ssl-hep/ServiceX_frontend/releases"
+"Releases" = "https://github.com/ssl-hep/ServiceX_frontend/releases"
+"Source Code" = "https://github.com/ssl-hep/ServiceX_frontend"
 
-[tool.poetry.extras]
-pandas = ["pandas", "pyarrow", "fastparquet"]
-docs = ["sphinx", "furo"]
+[project.optional-dependencies]
+pandas = [
+    "pandas>=2.0.2",
+    "pyarrow>=12.0.0",
+    "fastparquet>=2023.4.0",
+]
+# Developer extras
+test = [
+    "pytest>=7.2.0",
+    "pytest-mock>=3.10.0",
+    "pytest-cov>=4.0.0",
+    "flake8>=5.0.4",
+    "mypy>=0.981",
+    "pytest-asyncio>=0.21.0",
+    "asyncmock>=0.4.2",
+]
+docs = [
+    "sphinx>=7.0.1",
+    "furo>=2023.5.20",
+]
+develop = [
+    "servicex[pandas,test,docs]",
+]
 
+[tool.hatch.build.targets.sdist]
+# hatchling always includes:
+# pyproject.toml, .gitignore, any README, any LICENSE, AUTHORS
+include = [
+    "/servicex/",
+    "/tests/"
```

Comment:
@matthewfeickert - does this bundle the tests into the wheel? That seems unfriendly if so

Reply:
@BenGalewsky No it doesn't (and agreed that would be terrible)! `hatchling` builds the `sdist` from everything in the repo unless you tell it what you want with `include` or `only-include` and then it builds the wheel from the sdist, where it will only take the package contents.

Here's a full example of that:

```console
$ python -m build .
* Creating venv isolated environment...
* Installing packages in isolated environment... (hatchling>=1.13.0)
* Getting build dependencies for sdist...
* Building sdist...
* Building wheel from sdist
* Creating venv isolated environment...
* Installing packages in isolated environment... (hatchling>=1.13.0)
* Getting build dependencies for wheel...
* Building wheel...
Successfully built servicex-3.0.0a2.tar.gz and servicex-3.0.0a2-py3-none-any.whl
$ python -m tarfile --list dist/servicex-*.tar.gz
servicex-3.0.0a2/servicex/__init__.py 
servicex-3.0.0a2/servicex/_version.py 
servicex-3.0.0a2/servicex/configuration.py 
servicex-3.0.0a2/servicex/databinder_models.py 
servicex-3.0.0a2/servicex/dataset_group.py 
servicex-3.0.0a2/servicex/dataset_identifier.py 
servicex-3.0.0a2/servicex/expandable_progress.py 
servicex-3.0.0a2/servicex/minio_adapter.py 
servicex-3.0.0a2/servicex/models.py 
servicex-3.0.0a2/servicex/python_dataset.py 
servicex-3.0.0a2/servicex/query.py 
servicex-3.0.0a2/servicex/query_cache.py 
servicex-3.0.0a2/servicex/servicex_adapter.py 
servicex-3.0.0a2/servicex/servicex_client.py 
servicex-3.0.0a2/servicex/types.py 
servicex-3.0.0a2/servicex/app/__init__.py 
servicex-3.0.0a2/servicex/app/cache.py 
servicex-3.0.0a2/servicex/app/cli_options.py 
servicex-3.0.0a2/servicex/app/codegen.py 
servicex-3.0.0a2/servicex/app/main.py 
servicex-3.0.0a2/servicex/app/transforms.py 
servicex-3.0.0a2/servicex/databinder/__init__.py 
servicex-3.0.0a2/servicex/databinder/databinder.py 
servicex-3.0.0a2/servicex/databinder/databinder_configuration.py 
servicex-3.0.0a2/servicex/databinder/databinder_deliver.py 
servicex-3.0.0a2/servicex/databinder/databinder_outputs.py 
servicex-3.0.0a2/servicex/databinder/databinder_requests.py 
servicex-3.0.0a2/servicex/func_adl/__init__.py 
servicex-3.0.0a2/servicex/func_adl/func_adl_dataset.py 
servicex-3.0.0a2/servicex/func_adl/func_adl_dataset_group.py 
servicex-3.0.0a2/servicex/func_adl/util.py 
servicex-3.0.0a2/tests/__init__.py 
servicex-3.0.0a2/tests/conftest.py 
servicex-3.0.0a2/tests/example_config.yaml 
servicex-3.0.0a2/tests/example_config_no_cache_path.yaml 
servicex-3.0.0a2/tests/test_config.py 
servicex-3.0.0a2/tests/test_databinder.py 
servicex-3.0.0a2/tests/test_dataset.py 
servicex-3.0.0a2/tests/test_dataset_identifier.py 
servicex-3.0.0a2/tests/test_datest_group.py 
servicex-3.0.0a2/tests/test_expandable_progress.py 
servicex-3.0.0a2/tests/test_minio_adapter.py 
servicex-3.0.0a2/tests/test_python_dataset.py 
servicex-3.0.0a2/tests/test_query_cache.py 
servicex-3.0.0a2/tests/test_servicex_adapter.py 
servicex-3.0.0a2/tests/test_servicex_dataset.py 
servicex-3.0.0a2/.gitignore 
servicex-3.0.0a2/LICENSE 
servicex-3.0.0a2/README.md 
servicex-3.0.0a2/pyproject.toml 
servicex-3.0.0a2/PKG-INFO
$ python -m zipfile --list dist/servicex-*.whl
File Name                                             Modified             Size
servicex/__init__.py                           2020-02-02 00:00:00         2420
servicex/_version.py                           2020-02-02 00:00:00           96
servicex/configuration.py                      2020-02-02 00:00:00         5410
servicex/databinder_models.py                  2020-02-02 00:00:00         6270
servicex/dataset_group.py                      2020-02-02 00:00:00         3835
servicex/dataset_identifier.py                 2020-02-02 00:00:00         3403
servicex/expandable_progress.py                2020-02-02 00:00:00         8271
servicex/minio_adapter.py                      2020-02-02 00:00:00         4563
servicex/models.py                             2020-02-02 00:00:00         5232
servicex/python_dataset.py                     2020-02-02 00:00:00         3334
servicex/query.py                              2020-02-02 00:00:00        19972
servicex/query_cache.py                        2020-02-02 00:00:00         5189
servicex/servicex_adapter.py                   2020-02-02 00:00:00         5542
servicex/servicex_client.py                    2020-02-02 00:00:00        10921
servicex/types.py                              2020-02-02 00:00:00         1685
servicex/app/__init__.py                       2020-02-02 00:00:00         1535
servicex/app/cache.py                          2020-02-02 00:00:00         3350
servicex/app/cli_options.py                    2020-02-02 00:00:00         1776
servicex/app/codegen.py                        2020-02-02 00:00:00         2489
servicex/app/main.py                           2020-02-02 00:00:00         2295
servicex/app/transforms.py                     2020-02-02 00:00:00         5420
servicex/databinder/__init__.py                2020-02-02 00:00:00         1602
servicex/databinder/databinder.py              2020-02-02 00:00:00         2113
servicex/databinder/databinder_configuration.py 2020-02-02 00:00:00        10671
servicex/databinder/databinder_deliver.py      2020-02-02 00:00:00         3487
servicex/databinder/databinder_outputs.py      2020-02-02 00:00:00         3571
servicex/databinder/databinder_requests.py     2020-02-02 00:00:00         5585
servicex/func_adl/__init__.py                  2020-02-02 00:00:00         1535
servicex/func_adl/func_adl_dataset.py          2020-02-02 00:00:00        12484
servicex/func_adl/func_adl_dataset_group.py    2020-02-02 00:00:00         2438
servicex/func_adl/util.py                      2020-02-02 00:00:00         4013
servicex-3.0.0a2.dist-info/METADATA            2020-02-02 00:00:00         7544
servicex-3.0.0a2.dist-info/WHEEL               2020-02-02 00:00:00           87
servicex-3.0.0a2.dist-info/licenses/LICENSE    2020-02-02 00:00:00         1499
servicex-3.0.0a2.dist-info/RECORD              2020-02-02 00:00:00         2955
```